### PR TITLE
Improved Dockerfiles & added GitHub Actions

### LINKED ISSUE #1726: RPi Dockerfile has outdated glibc
### Bug Report Details ###
```
/ldc2-1.17.0-linux-armhf/bin/ldc2: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.28' not found (required by /ldc2-1.17.0-linux-armhf/bin/ldc2)
#15 3.410 
#15 3.643 configure: error: Compiler version insufficient, current compiler version , minimum version 1.17.0
```
[onedrive_rpi.log](https://github.com/abraunegg/onedrive/files/7598780/onedrive_rpi.log)

Setup: https://github.com/MarcelCoding/onedrive-docker/tree/f1dd8ae9e16a5e5d32eca05bed24d4e3966e3ff1

### PR DESCRIPTION
**Pull request by MarcelCoding**
When is the `contrib/docker/hooks/post_push` file executed? The GitHub action builds all Images and pushes on master and on a release. (Theoretically the script is no longer needed.)

The fedora image is taking over an hour to build, maybe that is a bit too long.
image

TODO for you:
* Add secret: `DOCKER_USERNAME` with value `driveone`
* Add secret: `DOCKER_TOKEN` with a token

Closes #1726

### CODE REVIEW

File: contrib/docker/Dockerfile.debian [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+RUN ARCH="$(dpkg --print-architecture)" \
+ && case "${ARCH}" in \
+      aarch64|arm64) \
+        BINARY_URL="https://github.com/ldc-developers/ldc/releases/download/v${LDC_VERSION}/ldc2-${LDC_VERSION}-linux-aarch64.tar.xz"; \
+        ;; \
+      armhf|arm) \
+        BINARY_URL="https://github.com/ldc-developers/ldc/releases/download/v${LDC_VERSION}/ldc2-${LDC_VERSION}-linux-armhf.tar.xz"; \
+        ;; \
+      amd64|i386:x86-64) \
+        BINARY_URL="https://github.com/ldc-developers/ldc/releases/download/v${LDC_VERSION}/ldc2-${LDC_VERSION}-linux-x86_64.tar.xz"; \
```
Comment (abraunegg) [inline contrib/docker/Dockerfile.debian:22]: @MarcelCoding 
The x86_64 version of LDC wont run on i386 | i686 .. and, in the past, there were many issues with i686 platforms due to compiler linkage. 

Can you drop i386 ?

**Comment by abraunegg** [comment]
@MarcelCoding 
Finally having a chance to review this. Many thanks for this effort and contribution.

File: contrib/docker/Dockerfile.debian (hunk 1 above)
Comment (MarcelCoding) [reply contrib/docker/Dockerfile.debian:22]: Sure, I actually tested x86 and noticed that it isn't working. I removed it from github actions and missed the Dockerfile.

**Comment by abraunegg** [comment]
@MarcelCoding 
> TODO for you:
> 
> * Add secret: `DOCKER_USERNAME` with value `driveone`
> * Add secret: `DOCKER_TOKEN` with a token

Unfortunately this did not work as intended - I suspect that this has something to do with that the 'docker secrets' are for the 'personal' account as you cannot create a Docker auth token for a 'free organisation' ...
docker_failure

Will have to look at this further in the new year as part of an inclusion to a later release.

Additionally, the [Docker.md](https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md) will need to be updated based on the newer/updated docker files

**Comment by MarcelCoding** [comment]
@abraunegg 
Ok, docker hub is not that cool anymore. I just checked my account and saw that: 
image
And in an orga: 
image

I think you have to create the access token inside of the account and not the organization. We could also drop docker hub and only use the GitHub container registry (ghcr.io). And sorry that I forgot the docker.md.

**Comment by abraunegg** [comment]
@MarcelCoding 
> I think you have to create the access token inside of the account and not the organization.

That is what I did - and still no go .. will have to look at this in great detail later

**Comment by abraunegg** [comment]
Closing this PR in favour of https://github.com/abraunegg/onedrive/pull/1795

**Pull request closed without merging**